homeassistant/components/amcrest/camera.py

```diff
@@ -144,8 +144,8 @@ async def async_setup_platform(
     registry = entity_registry.async_get(hass)
     old_entity_id = registry.async_get_entity_id(CAMERA_DOMAIN, DOMAIN, serial_number)
     if old_entity_id is not None:
-        _LOGGER.info(f"Updating unique id for camera {old_entity_id}")
-        await hass.async_add_executor_job(entity._update_unique_id)
+        _LOGGER.info("Updating unique id for camera %s", old_entity_id)
+        await hass.async_add_executor_job(entity.update_unique_id)
```

Comment:
this solves my question.

Reply:
Sure. I was thinking the unique id could be just computed in this function. It would duplicate where it is set between here and the `update` method, but it would simplify the `update` method to not have to call this function. If the unique id were something that is changing, then I feel like having the common helper is useful, but given that it shouldn't change and any change that we have will require some large migration like this, just duplicating the logic seems reasonable. What do you think about that?

Reply:
I agree with you.

Reply:
and btw I like your migration.

Reply:
Great, just pushed a new commit with the change here.